lib/oeqa/runtime/sanity/baseos.py

```diff
@@ -102,6 +102,16 @@ def test_baseos_systemd_process(self):
             status=1
         self.assertEqual(status, 0, msg="Error messages: %s" % output)
 
+    def test_baseos_systemd_boot_error(self):
+        ''' check systemd boot journal error''' 
+        known_issue_key_words="GPT" 
+        (status, output) = self.target.run("journalctl -ab")
+        self.assertEqual(status, 0, "Fail to run journalctl ")
+        self.assertTrue('' != output.strip(), "No systemd journal log")
+        (status, output) = self.target.run("journalctl -ab | grep -ivE '%s' | grep -i error " 
+                                           % known_issue_key_words )
+        self.assertTrue('' == output.strip(),"Errors in boot log")
```

Comment:
The test fails on some platforms, and the resulting error also shows that the test needs to be improved: when it fails, we want to see what the complete boot log is, otherwise it is impossible to diagnose the error without reproducing it locally (which is not always possible).

See https://ostroproject.org/jenkins/job/test_galileov2/847/testReport/junit/oeqa.runtime.sanity.baseos/BaseOsTest/test_baseos_systemd_boot_error/

Here's how I added additional output to a test failure:
https://github.com/01org/meta-intel-iot-security/commit/1df55345f273ce7e594b9ff1d27ce6de6adb6ce5

Instead of testing the result of the == comparison, better test the actual value: self.assertFalse(output.strip(), "Error in boot log:\n" + full_output.

Also missing are the enhancement discussed on IRC:
* make the grep -v more specific to the actual error
* check that journalctl really works by collecting its output without filtering and checking for non-empty. Also check the return code.